src/azure/cli/parser.py

```diff
@@ -62,11 +65,24 @@ def load_command_table(self, command_table):
                                                   help_file=metadata.help)
 
             argument_validators = []
+            argument_groups = {}
             for arg in metadata.arguments.values():
                 if arg.validator:
                     argument_validators.append(arg.validator)
-                param = command_parser.add_argument(
-                    *arg.options_list, **arg.options)
+                if arg.arg_group:
+                    try:
+                        group = argument_groups[arg.arg_group]
+                    except KeyError:
+                        # group not found so create
+                        name_comps = re.findall('[A-Z][^A-Z]*', arg.arg_group)
+                        group_name = '{} Arguments'.format(' '.join(name_comps))
```

Comment:
can this just print out the name of the arg_group instead of splitting and joining?